lib/topologies/mongos.js

```diff
@@ -1133,12 +1128,9 @@ Mongos.prototype.cursor = function(ns, cmd, options) {
  * @param {...object} param Parameters for the specific mechanism
  * @param {authResultCallback} callback A callback function
  */
-Mongos.prototype.auth = function(mechanism, db) {
-  var allArgs = Array.prototype.slice.call(arguments, 0).slice(0);
-  var self = this;
-  var args = Array.prototype.slice.call(arguments, 2);
-  var callback = args.pop();
-  var currentContextIndex = 0;
+Mongos.prototype.auth = function(credentials, callback) {
```

Comment:
documentation no longer lines up with the method signature

Reply:
also should we be doing validation of the parameters here?

Reply:
I don't think we should? Why would you `auth` with no credentials?

Reply:
🤷‍♂️ , I clearly have PTSD from our general lack of options validation 😓